import "ydb/public/api/protos/ydb_topic.proto";
package NKikimrPQClient;

message TKeyValue {
    optional string Key = 1;
    optional string Value = 2;
}

message TMapType {
    repeated TKeyValue Items = 1;
}

message TDataChunk {
    message TSessionHeader {
        optional string Server    = 1;
        optional string File      = 2;
        optional string Ident     = 3;
        optional string LogType   = 4;
    }

    optional TSessionHeader Meta = 124;

    // fileid in key
    optional uint64 SeqNo         = 1;
    optional uint64 CreateTime    = 3;   // chunk creation time in ms

    optional bytes Ip = 8;

    enum EChunkType {
        REGULAR = 0;
        GROW    = 1;
    }

    optional EChunkType ChunkType = 9 [default = REGULAR];

    optional int64 Codec = 10;

    optional TMapType ExtraFields     = 126;

    optional bytes Data           = 127; // ~ 64K

    repeated Ydb.Topic.MetadataItem MessageMeta = 128;
}
